File: 02-lletres/js/fase3.js
// Emmagatzemar en un Map tant les lletres de la llista com el nombre de vegades que apareixen
let nom = ['J', 'o', 'r', 'd', 'i'];

let mapa = new Map();

for (let index = 0; index < nom.length; index++) {
    const element = nom[index];

    if(mapa.has(element)){
        mapa.set(element, mapa.get(element) + 1);
    }else {
        mapa.set(element, 1);
    }
    
}

console.log(mapa);

File: 02-lletres/java/itinerari/variables/java/itinerari/variables/Fase3.java
package itinerari.variables;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Fase3 {

    // Emmagatzemar en un Map tant les lletres de la llista com el nombre de vegades
    // que apareixen
    public static Map<Character, Integer> executar(List<Character> llista) {

        Map<Character, Integer> mapa = new HashMap<Character, Integer>();

        for (Character character : llista) {
            if (!mapa.containsKey(Character.toLowerCase(character))) {
                mapa.put(character, 1);
            } else {
                mapa.put(character, mapa.get(character) + 1);
            }
        }

        return mapa;
    }

}